scripts/paper/generate_tablero_scaling_law.py

```
@@ -0,0 +1,382 @@
+        if row["timing_unit"] != "milliseconds":
+            raise SystemExit(f"{path} has unexpected timing_unit: {row['timing_unit']}")
+        if not parse_bool(row["verified"], label="verified"):
+            raise SystemExit(f"{path} contains unverified row at steps={row['steps']}")
+        metadata.add((row["timing_mode"], row["timing_policy"], row["timing_runs"]))
+        steps = parse_int(row["steps"], label="steps")
+        if steps & (steps - 1):
+            raise SystemExit(f"{path} has non-power-of-two step count: {steps}")
+        relation = row["relation"]
+        if relation == TYPED_RELATION:
+            if steps in typed:
+                raise SystemExit(f"{path} has duplicate typed row for steps={steps}")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)